### PR TITLE
Update guide and reference links in documentation README

### PR DESCRIPTION
**Pull request by mikewilburn**
This PR replaces, updates, or removes links which presently 404 due to the new Developer Site experience.

To simplify the process of reviewing this PR and understand the effect of the change, I've tried to make the commits as atomic as possible. In some cases, you can expand the commit message for more details.

### CODE REVIEW

**Comment by mikewilburn** [comment]
Hey @kerryrobinson & @esreli, I'm still looking for a few more replacement links, but feel free to begin reviewing and share any feedback on the changes I've made thus far.

Before I merge, I also plan to update the `RELEASE.md` doc and increment the release version number with a patch to account for this change. I'll eventually add a commit which accounts for that change inside of this PR.

File: docs/README.md [hunk 1]
```diff
@@ -215,22 +215,22 @@ Some [overlays](#overlays) get their information from local data sources, such a
 
 Specify the folders the app searches for local data using your [App configuration settings](#app-configuration-settings). Local data that has been found on your device is listed by this tool. Any data supported by ArcGIS Runtime SDKs can be used, including:
 
-- [GeoPackage]
+- [Geopackage]
 - [Tile package] (TPK)
 - [Mobile geodatabase]
 - KML
```
Comment (kerryrobinson) [inline docs/README.md:221]: I feel like we might as well add a link to KML in the glossary as well: https://developers.arcgis.com/documentation/glossary/kml/

**Review by kerryrobinson — COMMENTED** [review]
@mikewilburn so far the changes look good to me. frustrating some of the doc that is no longer in the guide that added some helpful context to developers.

File: docs/README.md (hunk 1 above)
Comment (mikewilburn) [reply docs/README.md]: Good suggestion - resolved with d0dae19

**Comment by mikewilburn** [comment]
@kerryrobinson completely agree. I think unfortunately, broken/removed links affecting this doc might be the norm and not the exception 😬 . Addressed that one comment you made - I think all should be good now.

@esreli I believe I've found and resolved all of the broken links now. This is ready for your review.

File: docs/README.md [hunk 2]
```diff
@@ -122,7 +122,7 @@ The navigation toolbar can be turned off in the [Settings panel](#settings-panel
 
 There is a compass on the display's lower-right corner. Its red pointer points to north. Touching the compass will rotate the map so the top points toward north.
 
-The [compass tool] is part of ArcGIS Runtime Toolkit for Qt.
+The [NorthArrow compass control] is part of ArcGIS Runtime Toolkit for Qt.
```
Comment (esreli) [inline docs/README.md:125]: Perhaps we link users to the Toolkit from this line?

**Review by esreli — COMMENTED** [review]
Great job capturing all of the dead links. I only have one general question and one in-line suggestion.

- There is a commented out image in the Compass section (line `120`). Can we resolve this?

**Comment by mikewilburn** [comment]
> - There is a commented out image in the Compass section (line 120). Can we resolve this?

Good spot. It's unclear to me why that was there. A little digging revealed that line 110 was the appropriate place for that, which is already the case. I've removed line 120 altogether.

File: docs/README.md (hunk 2 above)
Comment (mikewilburn) [reply docs/README.md]: I've left the specific reference to the NorthArrow component while still supplementing this line with a link to the Toolkit repo. Hopefully this is what you meant. I do think both links serve a purpose here.

**Comment by mikewilburn** [comment]
@esreli this PR now also includes an update to the `RELEASE.md` doc to describe the new documentation changes.

**Review by esreli — APPROVED** [review]
LGTM 👍

**Pull request merged**